cleanlab/experimental/tf_nlp.py

```diff
@@ -0,0 +1,74 @@
+"""
+A Scikeras classifier (adapted for huggingface models) which
+can be used for finding label issues in any nlp dataset.
+"""
+
+from scikeras.wrappers import KerasClassifier
+from sklearn.preprocessing import FunctionTransformer
+import tensorflow as tf
+import numpy as np
+
+
+class MultiInputKerasClassifier(KerasClassifier):
+    def __init__(self, train_input, seq_len, valid_input=None, y_val=None, **kwargs):
+        """
+        Basic scikeras/scikeras does not directly provide the option of having
+        a multi-input model. However, you can work around this problem by
+        adding the `feature_encoder` property to the class which extends
+        Scikitlearn's BaseEstimator.
+
+        References:
+        - https://towardsdatascience.com/scikeras-tutorial-a-multi-input-multi-output-wrapper-for-capsnet-hyperparameter-tuning-with-keras-3127690f7f28
+        - https://www.adriangb.com/scikeras/stable/notebooks/DataTransformers.html#4.-Multiple-inputs
+
+        """
+        super().__init__(**kwargs)
+        self.train_input = train_input
+        self.valid_input = valid_input
+        self.y_val = y_val
+        self.seq_len = seq_len
+
+    def split_input(self, X):
+        splitted_X = [
+            X[:, : self.seq_len],  # input_ids
+            X[:, self.seq_len :],  # attention_mask
+        ]
+        return splitted_X
+
+    @property
+    def feature_encoder(self):
+        return FunctionTransformer(
+            func=self.split_input,
+        )
+
+    def _get_tf_input(self, ids, train_input):
+        indexed_input_ids = np.array(tf.gather(train_input["input_ids"], indices=ids))
+
+        indexed_attention_mask = np.array(tf.gather(train_input["attention_mask"], indices=ids))
+
+        return np.hstack([indexed_input_ids, indexed_attention_mask])
+
+    def fit(self, ids, y, sample_weight=None):
```

Comment:
Documentation needed for args: ids, y, sample_weight

Reply:
Why did you decide to pass some data into the __init__() of this class and some data into fit()?  It seems nicer to pass all the data into fit() instead of __init__()

Reply:
Yes, I would have liked to do it that way, but when I call the `CleanLearning` fit I must pass a 2-D array as X. In this case the input is a 3-D like array. So what is happening now is that in the `__init__` the training dataset is loaded and as X of the `fit` I pass the ids corresponding to the data I want to use during the training. This is something that I don't like, but it is the only way that I found to make cleanlab work with multi-input data. If you have any suggestions to avoid this workaround I would appreciate it.

Reply:
Note that CleanLearning does support non-2D array data as demonstrated in this unit test:
https://github.com/cleanlab/cleanlab/blob/master/tests/test_classification.py#L567

But yes I see your point that it unfortunately does not support non-strictly array data and have opened an issue for this: 
https://github.com/cleanlab/cleanlab/issues/252

Thanks for pointing it out!  Let me know if you'd like to help with this :)

Reply:
Although the current way you had to get around this issue is not pretty, I also do not see an alternative work-around until we've added support for non-strictly array data in CleanLearning. If you want to address the other feedback on this PR, we should hopefully have this sorted out reasonably quickly.

Reply:
Happy to hear that you are working towards the integration of different input types in CleanLab, it would be really useful!

In the coming day I will fix the remaining suggestions. After that, we may discuss whether to use the non-strictly arrays within this or in a different PR.

Reply:
Sound good :) 
By then we should have fixed the issue hopefully and then you can avoid the current hack.